### PR TITLE
feat: adds drag and drop support to pages and docs

### LINKED ISSUE #355: [FR] Rearrange pages using drag-and-drop in the left-hand sidebar
**1~3 main use cases of the proposed feature**
Drag-and-drop pages make it easy to structure contents in the left-hand sidebar 

**what types of users can benefit from using your proposed feature**
Users who write a lot of pages

### PR DESCRIPTION
**Pull request by gaganyadav80**
Closes #355 

Overview:
- used flutters built-in `ReorderableListView.builder()` for pages section.
- used pub package called `Reorderables` for the need of `ReorderableColumn()` for docs section.

Both the pages and docs section uses a hash map to store the index for each item to persist it on hot reloads.
Pages use `appWidget[index].key.hashCode` and Docs use `views[index].id` to uniquely identify each item.

### CODE REVIEW

File: frontend/app_flowy/lib/workspace/presentation/home/menu/app/section/section.dart [hunk 1]
```diff
@@ -26,29 +29,111 @@ class ViewSection extends StatelessWidget {
...
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (views.isEmpty) {
+      _initItemList();
     }
 
-    return Column(children: viewWidgets);
+    log("BUILD: Section items: ${views.length}");
```
Comment (appflowy) [inline frontend/app_flowy/lib/workspace/presentation/home/menu/app/section/section.dart:104]: You could use Log.info

File: frontend/app_flowy/lib/workspace/presentation/home/menu/menu.dart [hunk 2]
```diff
@@ -40,13 +40,22 @@ class HomeMenu extends StatelessWidget {
...
         super(key: key);
 
+  @override
+  State<HomeMenu> createState() => _HomeMenuState();
+}
+
+class _HomeMenuState extends State<HomeMenu> {
+  /// Maps the hashmap of the menu items to their index in reorderable list view.
+  //TODO @gaganyadav80: Retain this map to persist on app restarts.
+  final Map<int, int> _menuItemIndex = <int, int>{};
```
Comment (appflowy) [inline frontend/app_flowy/lib/workspace/presentation/home/menu/menu.dart:50]: The order should be saved using the FlowySDK. I could implement that for you later. So there is no need to cover to StatefulWidget.

File: frontend/app_flowy/lib/workspace/presentation/home/menu/app/section/section.dart (hunk 1 above)
Comment (gaganyadav80) [reply frontend/app_flowy/lib/workspace/presentation/home/menu/app/section/section.dart:104]: I forgot to remove this but if I use this in future then I will keep it in mind. :)

**Comment by appflowy** [comment]
Hi, @gaganyadav80 I saw there were some conflicts with the upstream, do you have time to solve it or you can merge the upstream I can fix it for you?

**Comment by gaganyadav80** [comment]
> Hi, @gaganyadav80 I saw there were some conflicts with the upstream, do you have time to solve it or you can merge the upstream I can fix it for you?

I will solve it. :)

**Comment by gaganyadav80** [comment]
Hi @appflowy, I have resolved the merge conflicts.

**Pull request merged**